### PR TITLE
refactor get flag to throw open feature errors and handle targeting key error

### PR DESCRIPTION
**Pull request by Billlynch**
## Hi There, I just made a Pull Request!
Currently we set default value, error code and error message manually in the providers, this can make reading the getFlag function more difficult. OpenFeature's SDK will catch OF errors and map these in the exact same way. So we can simplify the logic with these errors. This is pre-work to make apply logic easier to adjust in the near future. 

Also added throwing invalid context error when confidence returns with a reason of targeting key error

#### :heavy_check_mark: Checklist

- [x] All tests are passing
- [ ] Relevant documentation updated
- [x] linter/style run on changed files
- [x] Tests added for new functionality
- [ ] Regression tests added for bug fixes
- [ ] Tested in a corresponding example app

### CODE REVIEW

File: packages/openfeature-server-provider/src/ConfidenceServerProvider.test.ts [hunk 1]
```diff
@@ -65,6 +72,13 @@ const dummyConfiguration: Configuration = {
       value: undefined,
       schema: 'undefined',
     },
+    ['targeting-error-flag']: {
+      name: 'targeting-error-flag',
```
Comment (fabriziodemaria) [inline packages/openfeature-server-provider/src/ConfidenceServerProvider.test.ts:76]: Perhaps `targeting-key-error-flag`?

File: packages/openfeature-web-provider/src/ConfidenceWebProvider.test.ts [hunk 2]
```diff
@@ -267,67 +296,34 @@ describe('ConfidenceProvider', () => {
...
-
-      expect(actual).toEqual({
-        value: false,
-        errorCode: ErrorCode.PROVIDER_NOT_READY,
-        reason: 'ERROR',
-      });
+    it('should throw if the provider is not ready', () => {
+      expect(() =>
+        instanceUnderTest.resolveBooleanEvaluation('testFlag.bool', false, dummyEvaluationContext, dummyConsole),
+      ).toThrow(new GeneralError('Provider not ready'));
```
Comment (fabriziodemaria) [inline packages/openfeature-web-provider/src/ConfidenceWebProvider.test.ts:302]: Seems like `PROVIDER_NOT_READY` [error](https://openfeature.dev/specification/types#error-code) has not been added [to the SDK](https://github.com/open-feature/js-sdk/tree/main/packages/shared/src/errors) yet. Perhaps we can open a PR in the SDK repo and then throw the related exception here?

**Review by fabriziodemaria — APPROVED** [review]
Approving, although to be more compliant to the OF specification we should probably address my comment?

File: packages/openfeature-web-provider/src/ConfidenceWebProvider.test.ts (hunk 2 above)
Comment (Billlynch) [reply packages/openfeature-web-provider/src/ConfidenceWebProvider.test.ts:302]: Yes, i have that change ready to go actually! :)

**Pull request closed without merging**